src/routes/Profile/Profile.tsx

```diff
@@ -35,11 +37,24 @@ const Profile: React.FunctionComponent = () => {
     ),
   );
 
+  const fetchBadges = async (seasonId: Season['id'], account: string) => {
+    return await getSeasonBadges({
+      userId: account,
+      seasonId,
+    });
+  };
+
   const getBadges = async (seasonId: Season['id'], account: string) => {
-    setLoading(true);
-    const result = await getSeasonBadges(account, seasonId);
+    const seasonUserId = getSeasonUserId(account, seasonId);
+    let result = getCacheValue(seasonUserId);
+
+    if (!result) {
+      setLoading(true);
+      result = await fetchBadges(seasonId, account);
+      setCacheValue(seasonUserId, result);
```

Comment:
so this function effectively caches the badges so that we don't need to retrieve the badges next time the user wants to renavigate to the profile page as long as they don't reload the app?

Reply:
well not really... As you can see on componentDidMount (useEffect with []) and when wallet changes I invalidate the cache. So this means that whenever the user will navigate between seasons using the SeasonToggler only then we will cache and fetch from cache

Reply:
got it, thanks